internal/api/v1/user/merkleroots/merklerootstest/merkleroots_sync_fixtures.go

```diff
@@ -0,0 +1,245 @@
+func sendJSONResponse(data any, w *http.ResponseWriter) {
+	(*w).Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(*w).Encode(data); err != nil {
+		(*w).WriteHeader(http.StatusInternalServerError)
+	}
+}
+
+// MockMerkleRootsAPIResponseNormal creates a mock server with normal API responses.
+func MockMerkleRootsAPIResponseNormal() *httptest.Server {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == "/api/v1/merkleroots" && r.Method == http.MethodGet {
+			lastEvaluatedKey := r.URL.Query().Get("lastEvaluatedKey")
+			sendJSONResponse(MockedMerkleRootsAPIResponseFn(lastEvaluatedKey), &w)
+		} else {
+			w.WriteHeader(http.StatusNotFound)
+		}
+	}))
+	return server
+}
+
+// MockMerkleRootsAPIResponseDelayed creates a mock server with delayed API responses.
+func MockMerkleRootsAPIResponseDelayed() *httptest.Server {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == "/api/v1/merkleroots" && r.Method == http.MethodGet {
+			lastEvaluatedKey := r.URL.Query().Get("lastEvaluatedKey")
+			all := MockedMerkleRootsAPIResponseFn(lastEvaluatedKey)
+			if len(all.Content) > 3 {
+				all.Content = all.Content[:3]
+			}
+			all.Page.Size = len(all.Content)
+			if len(all.Content) > 0 {
+				all.Page.LastEvaluatedKey = all.Content[len(all.Content)-1].MerkleRoot
+			} else {
+				all.Page.LastEvaluatedKey = ""
+			}
+			time.Sleep(50 * time.Millisecond)
+			sendJSONResponse(all, &w)
+		} else {
+			w.WriteHeader(http.StatusNotFound)
+		}
+	}))
+	return server
+}
+
+// MockMerkleRootsAPIResponseStale creates a mock server with a stale last evaluated key.
+func MockMerkleRootsAPIResponseStale() *httptest.Server {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == "/api/v1/merkleroots" && r.Method == http.MethodGet {
+			staleLastEvaluatedKeyResponse := models.ExclusiveStartKeyPage[[]models.MerkleRoot]{
+				Content: []models.MerkleRoot{
+					{
+						MerkleRoot:  "4a5e1e4baab89f3a32518a88c31bc87f618f76673e2cc77ab2127b7afdeda33b",
+						BlockHeight: 0,
+					},
+					{
+						MerkleRoot:  "0e3e2357e806b6cdb1f70b54c3a3a17b6714ee1f0e68bebb44a74b1efd512098",
+						BlockHeight: 1,
+					},
+					{
+						MerkleRoot:  "9b0fc92260312ce44e74ef369f5c66bbb85848f2eddd5a7a1cde251e54ccfdd5",
+						BlockHeight: 2,
+					},
+				},
+				Page: models.ExclusiveStartKeyPageInfo{
+					LastEvaluatedKey: "9b0fc92260312ce44e74ef369f5c66bbb85848f2eddd5a7a1cde251e54ccfdd5",
+					Size:             3,
+					TotalElements:    len(MockedSPVWalletData),
+				},
+			}
+			sendJSONResponse(staleLastEvaluatedKeyResponse, &w)
+		} else {
+			w.WriteHeader(http.StatusNotFound)
+		}
+	}))
+	return server
+}
+
+// MockedSPVWalletData is mocked  merkle roots data on spv-wallet side
+var MockedSPVWalletData = []models.MerkleRoot{
```

Comment:
Here the same as in the previous version I would extract these to some spv-wallet fixtures as this is in fact a fixture of what data spv wallet has rather than merkleroots fixture itself